### DISCUSSION TITLE
Test a private method?

### DISCUSSION THREAD

**Discussion by kozaka-tv** [Th0rstenf/Rocksniffer-to-Streamer.bot-extension#112] [General]
I learned that, "Test all the F time", and "never test a private" method! :)

If, to test the main method is too complex, too many test Data, etc., then we have to extract some codes into other classes to have them separated from the main method, and then test them there.

If this is not possible here with SB and so, then, in case we start to write test, the could be changed back to public. But I would awoid this.

_Originally posted by @kozaka-tv in https://github.com/Th0rstenf/Rocksniffer-to-Streamer.bot-extension/pull/93#discussion_r1165995551_

**Comment by Th0rstenf** [comment]
This was the main reason for splitting code into the subclasses. I agree that we should not get down to private method level, but the interface methods of subclasses should already provide a good entry. 
At some commit you changed a few back to private that I left public for testing, but we can check again once we actually start using a test framework.
